SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+#include "SpatialGDKFunctionalTests/SpatialGDK/RelevancyTest/RelevancyTest.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTest.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTestActor.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTestGameMode.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTestReplicatedActor.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTest.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTestActor.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTestReplicatedActor.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/SpatialTestPossession.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/SpatialTestRepossession.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/UNR-3066/OwnerOnlyPropertyReplication.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/UNR-3157/RPCInInterfaceTest.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestHandover/SpatialTestHandover.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/ReplicatedVisibilityTestActor.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/VisibilityTest.h"
+#include "Tests/AutomationEditorCommon.h"
+
+DEFINE_LOG_CATEGORY(LogGenerateTestMapsCommandlet);
+
+UGenerateTestMapsCommandlet::UGenerateTestMapsCommandlet()
+{
+	IsClient = false;
+	IsEditor = true;
+	IsServer = false;
+	LogToConsole = true;
+
+	ConstructorHelpers::FObjectFinder<UStaticMesh> PlaneAsset(TEXT("StaticMesh'/Engine/BasicShapes/Plane.Plane'"));
+	check(PlaneAsset.Succeeded());
+	PlaneStaticMesh = PlaneAsset.Object;
+
+	ConstructorHelpers::FObjectFinder<UMaterial> MaterialAsset(TEXT("Material'/Engine/BasicShapes/BasicShapeMaterial.BasicShapeMaterial'"));
+	check(MaterialAsset.Succeeded());
+	BasicShapeMaterial = MaterialAsset.Object;
+}
+
+UWorld* UGenerateTestMapsCommandlet::CreateNewTestMapWorld()
+{
+	UWorld* CurrentWorld = FAutomationEditorCommonUtils::CreateNewMap();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	AStaticMeshActor* Plane =
+		CastChecked<AStaticMeshActor>(GEditor->AddActor(CurrentLevel, AStaticMeshActor::StaticClass(), FTransform::Identity));
+	Plane->GetStaticMeshComponent()->SetStaticMesh(PlaneStaticMesh);
+	Plane->GetStaticMeshComponent()->SetMaterial(0, BasicShapeMaterial);
+	Plane->SetActorScale3D(FVector(
+		500, 500, 1)); // Make the initial platform larger so things don't fall off for tests which use a large area (visibility test)
+
+	AExponentialHeightFog* Fog =
```

Comment:
Is this a necessity?

Reply:
Just for visibility, so you can open the map and it looks okay.